docs/Events/listening-to-events.md

```diff
@@ -9,3 +9,72 @@ parent: "Events"
 ---
 
 # Listening to Events
+
+`bpmn-js` uses a general purpose [event bus](../Modules/common-modules/event-bus.md) provided by `diagram-js` to communicate across a diagram instance based on [`events`](./bpmn-events).
+
+## Retrieving the `eventBus` Instance
+
+In order to register a listener for an `event`, the `eventBus` instance needs to be retrieved first. This instance is registered as named diagram service in an internal class called `BaseViewer` and can therefore be retrieved using the `BaseViewer#get` method which is available in the sub-classes [`NavigatedViewer`](#), [`Viewer`](#) or [`Modeler`](#).
+
+Alternatively, the `#on` and `#off` methods can be used in [`NavigatedViewer`](#), [`Viewer`](#) or [`Modeler`](#) to directly register or remove a previously registered event listener.
+
+### Example Usage
+
+Notice that the following snippet demonstrates the retrieval of the `eventBus` instance using a [`Modeler`](#). In principle this instance could also be retrieved from an instance of a [`NavigatedViewer`](#) or [`Viewer`](#) likewise.
+```javascript
+import BpmnJS from 'bpmn-js/lib/Modeler';
+
+const modeler = new BpmnJS({
+  container,
+  keyboard: {
+    bindTo: document
+  }
+});
+
+// [...]
+
+const eventBus = modeler.get('eventBus');
+```
+
+
+## Registering an Event Listener
+
+[`EventBus`](../Modules/common-modules/event-bus.md) provides the [`EventBus#on`](#) and [`EventBus#once`](#) methods for registering events (see the reference pages for more details).
+
+### Example Usage
+
+Register a listener function with custom priority of `1500` (default is `1000`). A higher priority number results in a higher priority and therefore earlier calling).
+
+Notice that the following snippet could also use the [`EventBus#once`](#) method resulting in the listener function only being called once.
+```javascript
+function listener(event) {
+
+  // access event type
+  event.type; // 'saveSVG.start'
+
+  // stop propagation to other listeners
+  event.stopPropagation();
+
+  // prevent event default
+  event.preventDefault();
+}
+
+eventBus.on('saveSVG.start', 1500, listener);
```

Comment:
Documentation for returning a value is missing. You can do things like:

```javascript
eventBus.on('answerToEverything', () => 42);

const answerToEverything = eventBus.fire('answerToEverything');

console.log(answerToEverything); // 42
```

Reply:
I get that and would add that in the example. But I would understand the `listening-to-events` part primarily as an intro document. The documentation for returning values should rather be found in `event-bus.md` isn't it?

Reply:
I'd be alright with that, too. 👍